packages/gotrue/lib/src/gotrue_admin_api.dart

```diff
@@ -121,9 +121,20 @@ class GoTrueAdminApi {
   }
 
   /// Generates links to be sent via email or other.
+  ///
+  /// [password] is  used for [GenerateLinkType.signup]
+  ///
+  /// [newEmail] is used for [GenerateLinkType.emailChangeCurrent]
+  /// and [GenerateLinkType.emailChangeNew]
```

Comment:
I think we should do an assertion to ensure `newEmail` is not null if the type is `emailChangeCurrent` or `emailChangeNew`. Same with other properties.